dart/example/pubsub/stream.dart

```diff
@@ -0,0 +1,138 @@
+// Copyright (C) 2021 Toitware ApS.
+// Use of this source code is governed by a Zero-Clause BSD license that can
+// be found in the EXAMPLES_LICENSE file.
+
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:grpc/grpc.dart';
+
+import 'package:toit_api/toit/api/pubsub/subscribe.pbgrpc.dart';
+import 'package:toit_api/toit/model/pubsub/message.pb.dart';
+import 'package:uuid/uuid.dart';
+
+import 'subscription.dart';
+
+const String topic = "cloud:toit-api/example";
+
+/// Whether the given [error] is an error where the application should just retry.
+bool isRetryableGrpcError(GrpcError error) {
+  switch (error.code) {
+    case StatusCode.ok:
+    case StatusCode.unknown:
+    case StatusCode.deadlineExceeded:
+    case StatusCode.unavailable:
+      return true;
+
+    case StatusCode.cancelled:
+    case StatusCode.invalidArgument:
+    case StatusCode.notFound:
+    case StatusCode.alreadyExists:
+    case StatusCode.permissionDenied:
+    case StatusCode
+        .resourceExhausted: // Could also be acceptable for retry.case
+    case StatusCode.failedPrecondition:
+    case StatusCode.aborted:
+    case StatusCode.outOfRange:
+    case StatusCode.unimplemented:
+    case StatusCode.internal:
+    case StatusCode.dataLoss:
+    case StatusCode.unauthenticated:
+    default:
+      return false;
+  }
```

Comment:
IMO you only list the status codes that should be retried otherwise (`default:`) `return false`

Reply:
I think it's nicer to show that you thought about these errors.
You could still have forgotten one which falls under the `default`, but at least I have some idea of which errors won't be retried.